spinnaker-monitoring-daemon/spinnaker-monitoring/spectator_metric_transformer.py

```diff
@@ -25,15 +25,24 @@
 import sys
 import yaml
 
-
-# see https://stackoverflow.com/questions/1175208/\
-#     elegant-python-function-to-convert-camelcase-to-snake-case
-_SNAKE_CASE_RE = re.compile('((?<=[a-z0-9])[A-Z]|(?!^)(?<!_)[A-Z](?=[a-z]))')
-
+# This is a cache for the _snakeify function to increase performance on the
+#    conversion of CamelCase to snake_case
+SNAKEIFY_CACHE = {}
 
 def _snakeify(text):
-  """Turn text into snake-case."""
-  return _SNAKE_CASE_RE.sub(r'_\1', text).lower()
+  if text not in SNAKEIFY_CACHE.keys():
```

Comment:
Unless python is doing some short-circuiting here, this lookup will be `O(N)` in the number of keys in the cache, which will get worse and worse as the monitoring daemon has been running longer, potentially eliminating any benefit of caching.

Reply:
As the names and labels should be defined in the service code, and repeated on each lookup, this shouldn't be an issue.  I'd be happy to add some sort of aging out mechanism if it proves to be a problem.

Reply:
Looks like the option @ezimanyi suggested is the most performant, so used that one.